classes/class.ilMumieTaskGradeSync.php

```diff
@@ -136,24 +145,32 @@ private function getLastSync()
         if ($oldest_timestamp == PHP_INT_MAX) {
             $oldest_timestamp = 1;
         }
-        return $oldest_timestamp * 1000;
+        return $oldest_timestamp*1000;
     }
 
     /**
      * Get all users that can get marks for this MUMIE task
      */
     private function getAllUsers($task)
     {
-        global $ilDB;
-        $users = array();
-        $result = $ilDB->query("SELECT usr_id" .
-            " FROM ut_lp_marks" .
-            " WHERE obj_id = " . $ilDB->quote($task->getId(), "integer"));
+        if ($task->getParentRef() != 1) {
+            return ilParticipants::getInstance($task->getParentRef())->getMembers();
+        } else {
+            return $this->getAllUserIds();
+        }
+    }
 
-        while ($record = $ilDB->fetchAssoc($result)) {
-            array_push($users, $record['usr_id']);
+    public static function getAllUserIds()
+    {
+        global $ilDB;
+        $result = $ilDB->query(
+            "SELECT usr_id FROM usr_data;"
+        );
+        $allIds = array();
+        while ($user_id = $ilDB->fetchAssoc($result)) {
+            array_push($allIds, $user_id["usr_id"]);
         }
```

Comment:
i couldnt find a way to do it as array_map because the number of variables is the same as the amount of times fetchAssoc can be run before returning an error. Therefore the while loop would be needed anyways to count how many variables there are